tasks/test-metadata/0.3/test-metadata.yaml

```diff
@@ -0,0 +1,228 @@
+        echo "  GIT_REPOSITORY: $GIT_REPOSITORY"
+        echo "  GIT_URL: $GIT_URL"
+        echo "  GIT_REVISION: $GIT_REVISION"
+        echo "  PR_AUTHOR: $PR_AUTHOR"
+        echo "  COMPONENT_CONTAINER_IMAGE: $COMPONENT_CONTAINER_IMAGE"
+        echo "  SOURCE_REPO_URL: $SOURCE_REPO_URL"
+        echo "  SOURCE_REPO_BRANCH: $SOURCE_REPO_BRANCH"
+        echo "  TARGET_REPO_BRANCH: $TARGET_REPO_BRANCH"
+        echo "  SOURCE_REPO_ORG: $SOURCE_REPO_ORG"
+
+        # Write each environment variable to its respective result file (0.1 approach)
+        echo -n "$EVENT_TYPE" > $(results.test-event-type.path)
+        echo -n "$PULL_REQUEST_NUMBER" > $(results.pull-request-number.path)
+        echo -n "$GIT_ORGANIZATION" > $(results.git-org.path)
+        echo -n "$GIT_REPOSITORY" > $(results.git-repo.path)
+        echo -n "$COMPONENT_CONTAINER_IMAGE" > $(results.container-image.path)
+        echo -n "$GIT_URL" > $(results.git-url.path)
+        echo -n "$GIT_REVISION" > $(results.git-revision.path)
+        echo -n "$PR_AUTHOR" > $(results.pull-request-author.path)
+        echo -n "$JOB_SPEC" > $(results.job-spec.path)
+        echo -n "$SOURCE_REPO_URL" > $(results.source-repo-url.path)
+        echo -n "$SOURCE_REPO_BRANCH" > $(results.source-repo-branch.path)
+        echo -n "$TARGET_REPO_BRANCH" > $(results.target-repo-branch.path)
+        echo -n "$KONFLUX_COMPONENT_NAME" > $(results.component-name.path)
+
+        # Handle prefetch functionality (always enabled)
+        cosign download attestation "$COMPONENT_CONTAINER_IMAGE" > cosign_metadata.json 2>/dev/null || echo "{}" > cosign_metadata.json
+
+        CACHI2_SOURCE_ARTIFACT="$(jq -r \
+          '.payload | @base64d | fromjson | .predicate.buildConfig.tasks[] |
+          select(.name == "prefetch-dependencies") | .results[] | select(.name == "CACHI2_ARTIFACT") | .value' \
+          cosign_metadata.json 2>/dev/null || echo "")"
+
+        SOURCE_ARTIFACT="$(jq -r \
+          '.payload | @base64d | fromjson | .predicate.buildConfig.tasks[] |
+          select(.name == "prefetch-dependencies") | .results[] | select(.name == "SOURCE_ARTIFACT") | .value' \
+          cosign_metadata.json 2>/dev/null || echo "")"
+
+        # Check artifact status
+        if [[ -z "$CACHI2_SOURCE_ARTIFACT" || "$CACHI2_SOURCE_ARTIFACT" == "null" ]]; then
+          echo "[INFO] Cachi2 artifact: not found"
+          CACHI2_SOURCE_ARTIFACT=""
+        else
+          echo "[INFO] Cachi2 artifact: found ($CACHI2_SOURCE_ARTIFACT)"
+        fi
+
+        if [[ -z "$SOURCE_ARTIFACT" || "$SOURCE_ARTIFACT" == "null" ]]; then
+          echo "[INFO] Source artifact: not found"
+          SOURCE_ARTIFACT=""
+        else
+          echo "[INFO] Source artifact: found ($SOURCE_ARTIFACT)"
+        fi
+
+        # Extract repo and tag from image
+        IMAGE_WITHOUT_SHA="${COMPONENT_CONTAINER_IMAGE%%@sha256:*}"
+        CONTAINER_TAG="${IMAGE_WITHOUT_SHA##*:}"
```

Comment:
I'm not sure if CONTAINER_TAG works in any case. I extracted a container image from some snapshot which is in this format
`quay.io/redhat-user-workloads/rhtap-integration-tenant/integration-service/integration-service@sha256:b73a6fbc8eaa59d0c4751f8d082e5735cda27a544091c8a59e738fa249b91e5b`

and after removing the sha256 suffix, no tag remains... so it seems like this line is not needed?